src/ConfigR/BasicConfig.Dictionary.cs

```diff
@@ -57,7 +59,7 @@ public bool ContainsKey(string key)
 
         public bool Remove(string key)
         {
-            log.TraceFormat(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource());
+            log.TraceFormat(string.Format(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource()));

```

Comment:
You don't need to do lazy loading if you remove the call to `string.Format()`, since then the arguments will be passed straight through to LibLog and it will only use them if the logger is enabled.